## <center>Практическое задание к лекции 2.

**1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.**

Для решения данной задачи воспользуемся формулой **Бернулли**: $\displaystyle P_{n}(X=k)=C^k_{n}p^{k}q^{n-k}$,  
где $n=100\\k=85\\p=0.8\\q=0.2$ тогда  
$P_{100}(X=85)=C^{85}_{100}\cdot0.8^{85}\cdot0.2^{15}=\frac{100!}{85!(100-85)!}\cdot0.8^{85}\cdot0.2^{15}$   
Вычислим:

In [21]:
from math import factorial, exp
def bernulli(n, k, p):
    comb=factorial(n)/(factorial(k)*factorial(n-k))
    return comb*(p**k)*(1-p)**(n-k)

In [3]:
print(f'P = {bernulli(100,85,0.8): .4f}')

P =  0.0481


---  

#### Ответ:  
Вероятность того, что стрелок попадет в цель ровно 85 раз: $P=0.0481=4.81\%$  

---

**2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек.**  
   **a) Какова вероятность, что ни одна из них не перегорит в первый день?**  
   **б) Какова вероятность, что перегорят ровно две?**  

Поскольку в задаче вероятность наступления события мала, а количество испытаний велико, для решения задачи воспользуемся формулой **Пуассона**:  

$\displaystyle P_m=\frac{\lambda^m}{m!}\cdot e^{-\lambda}\text{, где }\lambda \text{ среднее количество наступления события за определенную единицу измерений, и }\lambda=p\cdot n$
Для облегчения расчётов напишем функцию:  

In [14]:
def poisson_distr(m,p,n):
    lamb=p*n
    return exp(-lamb)*(lamb**m)/factorial(m)

a) По условию задания:
$\displaystyle p=0.0004\\n=5000\\m=0$  
тогда:

In [16]:
print(f'P = {poisson_distr(0,0.0004,5000): .4f}')

P =  0.1353


---  
#### Ответ:  
Вероятность того, что ни одна из ламп не перегорит в первый день $\displaystyle P=0.1353=0.1353\%$  

---

б) По условию задания:  
$\displaystyle p=0.0004\\n=5000\\m=2$  
тогда:

In [17]:
print(f'P = {poisson_distr(2,0.0004,5000): .4f}')

P =  0.2707


---  
#### Ответ:  
Вероятность того, что 2 лампы перегорят в первый день $\displaystyle P=0.2707=0.2707\%$  

---

**3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?**

Воспользуемся формулой **Бернулли**, 
в нашем случае:  
$n=144\\k=70\\p=0.5$  
Вычислим:  

In [22]:
print(f'P = {bernulli(144,70,0.5): .4f}')

P =  0.0628


---  
#### Ответ:  
Вероятность того, что при 144-х кратном подбрасывании монетки орёл выпадет ровно 70 раз $\displaystyle P=0.0628=0.0628\%$  

---

**4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что:**  
   **a) все мячи белые?;**  
   **б) ровно два мяча белые?;**  
   **в) хотя бы один мяч белый?.**

Для решения данной задачи подсчитаем общее число элементарных исходов при вытаскивании мячей:  
Для первой корзины это $n_1=C^2_{10}$    
Для второй $n_2=C^2_{11}$  
Напишем функцию, для подсчёта количества комбинаций:

In [23]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Вычислим $n_1$ и $n_2$:

In [44]:
n1=combinations(10,2)
n2=combinations(11,2)
print(f'Общее число элементарных исходов для первой корзины: n1 = {n1}\n'
f'Общее число элементарных исходов для второй корзины: n2 = {n2}')

Общее число элементарных исходов для первой корзины: n1 = 45
Общее число элементарных исходов для второй корзины: n2 = 55


a) Для выполнения этого условия необходимо **одновременное** наступление двух событий:  
событие $A$-из первой корзины извлекли 2 белых шара,  
событие $B$-из второй корзины извлекли 2 белых шара.  
Для события $A$ число благоприятных элементарных исходов $m_1=C^2_{7}$  
Для события $B$ число благоприятных элементарных исходов $m_2=C^2_{9}$  
Вычислим $m_1$ и $m_2$:

In [45]:
m1=combinations(7,2)
m2=combinations(9,2)
print(f'Число благоприятных элементарных исходов для события А: m1 = {combinations(7,2)}\n'
      f'Число благоприятных элементарных исходов для события В: m2 = {combinations(9,2)}'
     )

Число благоприятных элементарных исходов для события А: m1 = 21
Число благоприятных элементарных исходов для события В: m2 = 36


Для событий $A$ и $B$ вероятности их наступления будут расчитываться как $P(A)=\frac{m_1}{n_1}$ и $P(B)=\frac{m_2}{n_2}$, а вероятность их одновременного наступления как $P(A+B)=P(A)\cdot P(B)$  
Расчитаем:

In [46]:
print(f'Вероятность наступления события А, Р(А) = {m1/n1: .4f}\n'
      f'Вероятность наступления события В, Р(В) = {m2/n2: .4f}\n'
      f'Вероятность одновременного наступления событий А и В, Р(А+В) = {(m1/n1)*(m2/n2): .4f}'
     )

Вероятность наступления события А, Р(А) =  0.4667
Вероятность наступления события В, Р(В) =  0.6545
Вероятность одновременного наступления событий А и В, Р(А+В) =  0.3055


$* \text{Вариант решения через условную вероятность:}\\P(A+B)=\frac{7}{10}\cdot\frac{6}{9}\cdot\frac{9}{11}\cdot\frac{8}{10}$

In [47]:
P=(7/10)*(6/9)*(9/11)*(8/10)
print(f'Вероятность одновременного наступления событий А и В, Р(А+В) = {P: .4f}')

Вероятность одновременного наступления событий А и В, Р(А+В) =  0.3055


---  
#### Ответ:  
Вероятность того, что все извлечённые мячи белые: $P\approx0.3055\approx30.55\%$  

---

б) Этому варианту соответствует несколько комбинаций:  
   1. События $A$-из первой корзины извлекли 2 белых шара **и** $B$-из второй корзины извлекли 2 не белых шара, **или**  
   2. События $C$-из первой корзины извлекли 1 белый и 1 не белый шар **и** $D$-из второй корзины извлекли 1 белый и 1 не белый шар, **или**  
   3. События $E$-из первой корзины извлекли 2 не белых шара **и** $F$-из второй корзины извлекли 2 белых шара  
   
А вероятность того, что сложится **одна из** этих комбинаций, будет суммой вероятностей этих комбинаций. Т.е.:  
$P=P(A)\cdot P(B)+P(C)\cdot P(D)+P(E)\cdot P(F)$  
Число **благоприятных** элементарных исходов для каждого события в комбинации:  
   1. $m_A=C^2_{7}$, $m_B=C^0_{9}$;  
   2. $m_C=C^1_{7}\cdot C^1_3$, $m_D=C^1_{9}\cdot C^1_2$;  
   3. $m_E=C^0_{7}$, $m_F=C^2_{9}$.  

$\Rightarrow P=\frac{m_A}{n1}\cdot\frac{m_B}{n2}+\frac{m_C}{n1}\cdot\frac{m_D}{n2}+\frac{m_E}{n1}\cdot\frac{m_F}{n2}=\frac{C^2_{7}}{n1}\cdot\frac{C^0_{9}}{n2}+\frac{C^1_{7}\cdot C^1_3}{n1}\cdot\frac{C^1_{9}\cdot C^1_2}{n2}+\frac{C^0_{7}}{n1}\cdot\frac{C^2_{9}}{n2}$  
Произведём вычисления:

In [51]:
m_A=combinations(7,2)
m_B=combinations(2,2)
m_C=combinations(7,1)*combinations(3,1)
m_D=combinations(9,1)*combinations(2,1)
m_E=combinations(3,2)
m_F=combinations(9,2)
print(f'Число благоприятных элементарных исходов для события А: m_A = {m_A}\n'
      f'Число благоприятных элементарных исходов для события B: m_B = {m_B}\n'
      f'Число благоприятных элементарных исходов для события C: m_C = {m_C}\n'
      f'Число благоприятных элементарных исходов для события D: m_D = {m_D}\n'
      f'Число благоприятных элементарных исходов для события E: m_E = {m_E}\n'
      f'Число благоприятных элементарных исходов для события F: m_F = {m_F}'  
     )

Число благоприятных элементарных исходов для события А: m_A = 21
Число благоприятных элементарных исходов для события B: m_B = 1
Число благоприятных элементарных исходов для события C: m_C = 21
Число благоприятных элементарных исходов для события D: m_D = 18
Число благоприятных элементарных исходов для события E: m_E = 3
Число благоприятных элементарных исходов для события F: m_F = 36


In [52]:
P=(m_A/n1)*(m_B/n2)+(m_C/n1)*(m_D/n2)+(m_E/n1)*(m_F/n2)
print(f'Вероятность того, что будут вытащены ровно 2 белых мяча Р = {P: .4f}')

Вероятность того, что будут вытащены ровно 2 белых мяча Р =  0.2048


$* \text{Вариант решения через условную вероятность:}\\P=\frac{7}{10}\cdot\frac{6}{9}\cdot\frac{2}{11}\cdot\frac{1}{10}+4^*\cdot\left(\frac{7}{10}\cdot\frac{3}{9}\cdot\frac{9}{11}\cdot\frac{2}{10}\right)+\frac{3}{10}\cdot\frac{2}{9}\cdot\frac{9}{11}\cdot\frac{8}{10}$  
\* *Второе слагаемое нужно умножить на **4**, поскольку существует 4 способа достать из каждой корзины 1 белый и 1 не белый мяч.*

In [57]:
P1=((7/10)*(6/9)*(2/11)*(1/10))+4*((7/10)*(3/9)*(9/11)*(2/10))+((3/10)*(2/9)*(9/11)*(8/10))
print(f'Вероятность того, что будут вытащены ровно 2 белых мяча Р = {P1: .4f}')

Вероятность того, что будут вытащены ровно 2 белых мяча Р =  0.2048


---  
#### Ответ:  
Вероятность того, что будут вытащены ровно 2 белых мяча: $P\approx0.2048\approx20.48\%$  

---

в) В данном случае удобнее рассмотреть событие $\bar A$-"не было извлечено **ни одного** белого мяча", которое наряду с искомым событием $A$-"был извлечен хотя бы 1 белый мяч", составляют **полную группу** событий, по скольку в результате испытаний может произойти **только** либо событие $A$, либо событие $\bar A$, и поэтому:  
$P(A)=1-P(\bar A)$  
Для наступления события $\bar A$ необходимо чтобы из первой и второй корзин не было вынуто ни одного белого мяча, число таких благоприятных исходов для первой корзины $m_1=C^2_3$ а для второй $C^2_2$  
Вычислим $m_1$ и $m_2$:

In [59]:
m1=combinations(3,2)
m2=combinations(2,2)
print(f'Число благоприятных элементарных исходов для первой корзины: m1 = {combinations(3,2)}\n'
      f'Число благоприятных элементарных исходов для второй корзины: m2 = {combinations(2,2)}'
     )

Число благоприятных элементарных исходов для первой корзины: m1 = 3
Число благоприятных элементарных исходов для второй корзины: m2 = 1


Тогда $P(\bar A)=\frac{m1}{n1}\cdot\frac{m2}{n2}$, вычислим $P(\bar A)$ и $P(A)$:

In [61]:
print(f'Вероятность того, что не будет извлечено ни одного белого мяча Р(А_обр) = {(m1/n1)*(m2/n2): .4f}\n'
      f'Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = 1 - Р(А_обр) = {1-(m1/n1)*(m2/n2): .4f}'
     )

Вероятность того, что не будет извлечено ни одного белого мяча Р(А_обр) =  0.0012
Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = 1 - Р(А_обр) =  0.9988


$* \text{Вариант решения через условную вероятность:}\\P(A)=1-\frac{3}{10}\cdot\frac{2}{9}\cdot\frac{2}{11}\cdot\frac{1}{10}$

In [63]:
P=1-(3/10)*(2/9)*(2/11)*(1/10)
print(f'Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = {P: .4f}')

Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) =  0.9988


---  
#### Ответ:  
Вероятность того, что будет извлечен хотя бы один белый мяч $P(A)\approx0.9988\approx99.88\%$  

---